.infrastructure/helm-chart/todoapp/charts/mysql/templates/secret.yml

```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ .Chart.Name }}-secrets
+  namespace: {{ .Values.mysql.namespace }}
+type: Opaque
+data:
+  {{- range $k, $v := .Values.mysql.secret }}
+    {{ $k }}: {{ $v }}
+  {{ end }}
```

Comment:
Ensure that the `mysql.secret` values are defined in the `values.yaml` file. This is necessary for populating the Secret data correctly.